opsdroid/database/matrix/__init__.py

```diff
@@ -0,0 +1,249 @@
+"""Database that uses the matrix connector."""
+
+import logging
+from contextlib import contextmanager
+
+from nio import RoomGetStateError, RoomGetStateEventError
+from opsdroid.database import Database
+from opsdroid.connector.matrix.events import MatrixStateEvent, GenericMatrixRoomEvent
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DatabaseMatrix(Database):
+    """A module for opsdroid to allow memory to persist in matrix room state."""
+
+    def __init__(self, config, opsdroid=None):
+        """Start the database connection."""
+        super().__init__(config, opsdroid=opsdroid)
+        self.name = "matrix"
+        self.room = config.get("default_room", "main")
+        self._single_state_key = config.get("single_state_key", True)
+        self.should_encrypt = config.get("should_encrypt", False)
+        self._event_type = "dev.opsdroid.database"
+        self.should_migrate = True
+
+        _LOGGER.debug("Loaded matrix database connector.")
+
+    @property
+    def connector(self):
+        """Define the initiated matrix connector."""
+        return self.opsdroid.get_connector("matrix")
+
+    @property
+    def room_id(self):
+        """Define ID of the room to store data in."""
+        return (
```

Comment:
Could we use the connector `lookup_target` method here? https://github.com/opsdroid/opsdroid/blob/master/opsdroid/connector/matrix/connector.py#L294